### PR TITLE
[Refactor](pipeline) refactor the pipeline exec code remove unless code

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

1. remove the unless timer in pipeline fragment context
2. remove the unless code in local shuffle
3. change tuple to struct

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run be_ut

File: be/src/pipeline/local_exchange/local_exchanger.cpp [hunk 1]
```diff
@@ -46,36 +46,31 @@ Status ShuffleExchanger::get_block(RuntimeState* state, vectorized::Block* block
...
-            mutable_block = vectorized::MutableBlock::create_unique(
-                    partitioned_block.first->data_block.clone_empty());
-            get_data(block);
-        } else {
-            COUNTER_UPDATE(local_state._get_block_failed_counter, 1);
-            *eos = true;
-        }
-    } else if (_data_queue[local_state._channel_id].try_dequeue(partitioned_block)) {
+
+    if (_data_queue[local_state._channel_id].try_dequeue(partitioned_block)) {
```
Comment (Gabriel39) [inline be/src/pipeline/local_exchange/local_exchanger.cpp:62]: Assume the following events occur in order:
1. Source operator `_data_queue[local_state._channel_id].try_dequeue(partitioned_block)` return false.
2. The last sink operator push the last block into `_data_queue`.
3. The last sink operator sub `_running_sink_operators` to zero.
4. Source operator thinks `_running_sink_operators` is true.

And then, eos is true because of event 4. So we ignore this last block and its results will be incorrect.

File: be/src/pipeline/local_exchange/local_exchanger.cpp [hunk 2]
```diff
@@ -186,23 +180,17 @@ Status PassthroughExchanger::sink(RuntimeState* state, vectorized::Block* in_blo
...
-            block->swap(next_block);
-            _free_blocks.enqueue(std::move(next_block));
-            local_state._shared_state->sub_mem_usage(local_state._channel_id,
-                                                     block->allocated_bytes());
-        } else {
-            COUNTER_UPDATE(local_state._get_block_failed_counter, 1);
-            *eos = true;
-        }
-    } else if (_data_queue[local_state._channel_id].try_dequeue(next_block)) {
+    if (_data_queue[local_state._channel_id].try_dequeue(next_block)) {
```
Comment (Gabriel39) [inline be/src/pipeline/local_exchange/local_exchanger.cpp:183]: ditto

File: be/src/pipeline/local_exchange/local_exchanger.cpp [hunk 3]
```diff
@@ -361,23 +343,17 @@ Status AdaptivePassthroughExchanger::get_block(RuntimeState* state, vectorized::
...
```
Comment (Gabriel39) [inline be/src/pipeline/local_exchange/local_exchanger.cpp:346]: ditto

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Review by Gabriel39 — DISMISSED** [review]

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

File: be/src/pipeline/pipeline_task.cpp [hunk 4]
```diff
@@ -251,7 +250,6 @@ Status PipelineTask::execute(bool* eos) {
             COUNTER_UPDATE(_yield_counts, 1);
             break;
         }
-        SCOPED_RAW_TIMER(&time_spent);
```
Comment (mengzhisy) [inline be/src/pipeline/pipeline_task.cpp:254]: Why was this line removed? It seems to have broken the dynamic priority level adjustment of PipelineTask in task_queue.
